src/densities.jl

```diff
@@ -76,33 +79,42 @@ is not collinear the spin density is `nothing`.
     end
 
     Threads.@threads for (ikpts, ρaccu) in collect(zip(kpt_per_thread, ρaccus))
-        for iσ in 1:n_spin
-            ρaccu[iσ] .= 0
-        end
+        ρaccu .= 0
         ρ_k = similar(ψ[1][:, 1], basis.fft_size)
         for ik in ikpts
             kpt = basis.kpoints[ik]
             compute_partial_density!(ρ_k, basis, kpt, ψ[ik], occupation[ik])
             lowpass_for_symmetry!(ρ_k, basis)
             # accumulates all the symops of ρ_k into ρaccu
-            accumulate_over_symmetries!(ρaccu[kpt.spin], ρ_k, basis, basis.ksymops[ik])
+            accumulate_over_symmetries!(ρaccu[:, :, :, kpt.spin], ρ_k, basis, basis.ksymops[ik])
         end
     end
 
     # Count the number of k-points modulo spin
     count = sum(length(basis.ksymops[ik]) for ik in 1:length(basis.kpoints)) ÷ n_spin
     count = mpi_sum(count, basis.comm_kpts)
-    ρs = [sum(getindex.(ρaccus, iσ)) / count for iσ in 1:n_spin]
-    mpi_sum!.(ρs, Ref(basis.comm_kpts))
+    ρ = sum(ρaccus) ./ count
+    mpi_sum!(ρ, basis.comm_kpts)
+    G_to_r(basis, ρ)
+end
 
-    @assert basis.model.spin_polarization in (:none, :spinless, :collinear)
-    if basis.model.spin_polarization == :collinear
-        ρtot  = from_fourier(basis, ρs[1] + ρs[2])  # up + down
-        ρspin = from_fourier(basis, ρs[1] - ρs[2])  # up - down
+total_density(ρ) = dropdims(sum(ρ; dims=4); dims=4)
+@views function spin_density(ρ)
+    if size(ρ, 4) == 2
+        ρ[:, :, :, 1] - ρ[:, :, :, 2]
     else
-        ρtot  = from_fourier(basis, ρs[1])
-        ρspin = nothing
+        0*ρ[:, :, :]
     end
+end
 
-    ρtot, ρspin
+function ρ_from_total_and_spin(ρtot, ρspin)
```

Comment:
Why not just `from_total_and_spin`?

Reply:
how do you know you're talking about densities and not wf/potentials?

Reply:
That's kind of the point. It should work for densities and potentials the same way, shouldn't it?